flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceUtils.java

```diff
@@ -551,10 +551,16 @@ private static void sanityCheckShuffleMemory(final Configuration config, final M
 					+ shuffleRangeFraction.maxSize.toString() + "].");
 			}
 			if (isShuffleMemoryFractionExplicitlyConfigured(config) &&
-				!derivedShuffleMemorySize.equals(totalFlinkMemorySize.multiply(shuffleRangeFraction.fraction))) {
-				throw new IllegalConfigurationException("Derived Shuffle Memory size("
-					+ derivedShuffleMemorySize.toString() + ") does not match configured Shuffle Memory fraction ("
-					+ shuffleRangeFraction.fraction + ").");
+				!derivedShuffleMemorySize.equals(totalFlinkMemorySize.multiply(shuffleRangeFraction.fraction)) &&
+				derivedShuffleMemorySize.getBytes() < shuffleRangeFraction.maxSize.getBytes() &&
+				derivedShuffleMemorySize.getBytes() > shuffleRangeFraction.minSize.getBytes()) {
```

Comment:
I would factor the range checks out into `isContainedInRange(memory, min, max)`. Maybe this can be reused somewhere else.